src/ip2gid.c

```diff
@@ -0,0 +1,86 @@
+/* SPDX-License-Identifier: ((GPL-2.0 WITH Linux-syscall-note) OR BSD-2-Clause) */
```

Comment:
ip2gid.c is a bad name for this.

What's the plan, do you want to group all msg validation in one file?|
so name the file to reflect that.